examples/pmp_convor.py

```diff
@@ -0,0 +1,212 @@
+data = data[:20]
+
+_ = plot_images(data[:, 0], nr=4)
+
+# %% [markdown]
+# ### Construct variable grid, initialize factor graph, and add factors
+
+# %% [markdown]
+# We build a factor graph equivalent to the one in the PMP paper.
+
+# %%
+n_images, n_chan, im_height, im_width = data.shape
+n_feat, feat_height, feat_width = 5, 6, 6
+s_height = im_height - feat_height + 1
+s_width = im_width - feat_width + 1
+
+S = groups.NDVariableArray(num_states=2, shape=(n_images, n_feat, s_height, s_width))
+W = groups.NDVariableArray(
+    num_states=2, shape=(n_chan, n_feat, feat_height, feat_width)
+)
+SW = groups.NDVariableArray(
+    num_states=2,
+    shape=(n_images, n_chan, im_height, im_width, n_feat, feat_height, feat_width),
+)
+X = groups.NDVariableArray(num_states=2, shape=data.shape)
+
+fg = graph.FactorGraph(variables=dict(S=S, W=W, SW=SW, X=X))
+
+# %%
+variable_names_for_OR_factors = {}
+
+# Add ANDFactors
+for idx_img in tqdm(range(n_images)):
+    for idx_chan in range(n_chan):
+        for idx_s_height in range(s_height):
+            for idx_s_width in range(s_width):
+                for idx_feat in range(n_feat):
+                    for idx_feat_height in range(feat_height):
+                        for idx_feat_width in range(feat_width):
+                            idx_img_height = idx_feat_height + idx_s_height
+                            idx_img_width = idx_feat_width + idx_s_width
+                            SW_var = (
+                                "SW",
+                                idx_img,
+                                idx_chan,
+                                idx_img_height,
+                                idx_img_width,
+                                idx_feat,
+                                idx_feat_height,
+                                idx_feat_width,
+                            )
+
+                            variable_names = [
+                                ("S", idx_img, idx_feat, idx_s_height, idx_s_width),
+                                (
+                                    "W",
+                                    idx_chan,
+                                    idx_feat,
+                                    idx_feat_height,
+                                    idx_feat_width,
+                                ),
+                                SW_var,
+                            ]
+
+                            fg.add_factor_by_type(
+                                variable_names=variable_names,
+                                factor_type=logical.ANDFactor,
+                            )
+
+                            X_var = (idx_img, idx_chan, idx_img_height, idx_img_width)
+                            if X_var not in variable_names_for_OR_factors:
+                                variable_names_for_OR_factors[X_var] = [SW_var]
+                            else:
+                                variable_names_for_OR_factors[X_var].append(SW_var)
+
+
+# Add ORFactors
```

Comment:
Make this comment more descriptive, how the ORFactors combine different SW variables as parents of pixel variables